R/pkg/R/deserialize.R

```diff
@@ -234,24 +234,27 @@ readMultipleObjectsWithKeys <- function(inputCon) {
 readDeserializeInArrow <- function(inputCon) {
   # This is a hack to avoid CRAN check. Arrow is not uploaded into CRAN now. See ARROW-3204.
   requireNamespace1 <- requireNamespace
-  requireNamespace1("arrow", quietly = TRUE)
-  RecordBatchStreamReader <- get(
-    "RecordBatchStreamReader", envir = asNamespace("arrow"), inherits = FALSE)
-  as_tibble <- get("as_tibble", envir = asNamespace("arrow"))
+  if (requireNamespace1("arrow", quietly = TRUE)) {
```

Comment:
another approach
```
  if (!requireNamespace1("arrow", quietly = TRUE)) {
    stop("'arrow' package should be installed.")
  }
```

Reply:
Let me fix it when I clean up this logic to be same and minimised.